src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts.expected

```diff
@@ -0,0 +1,89 @@
+-----------------
+INPUT
+----------------- 
+/** @gqlInterface */
+interface Entity {
+  name: string | null;
+}
+
+/** @gqlField */
+export function greeting(entity: Entity): string {
+  return `Hello, ${entity.name ?? "World"}!`;
+}
+
+/** @gqlInterface */
+interface IThing extends Entity {
+  name: string | null;
+}
+
+/** @gqlType */
+export class Doohickey implements IThing, Entity {
+  __typename: "Doohickey";
+  name: string;
+}
+
+-----------------
+OUTPUT
+-----------------
+-- SDL --
+interface Entity {
+  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
+}
+
+interface IThing implements Entity {
+  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
+}
+
+type Doohickey implements Entity & IThing {
+  greeting: String @metadata(argCount: 1, name: "greeting", tsModulePath: "grats/src/tests/fixtures/interfaces/concreteTypeInheritsFunctionFieldFromTransitiveInterfacets.ts")
+}
+-- TypeScript --
+import { greeting as doohickeyGreetingResolver } from "./concreteTypeInheritsFunctionFieldFromTransitiveInterfacets";
+import { GraphQLSchema, GraphQLInterfaceType, GraphQLString, GraphQLObjectType } from "graphql";
+export function getSchema(): GraphQLSchema {
+    const EntityType: GraphQLInterfaceType = new GraphQLInterfaceType({
+        name: "Entity",
+        fields() {
+            return {
+                greeting: {
+                    name: "greeting",
+                    type: GraphQLString
+                }
+            };
+        }
+    });
+    const IThingType: GraphQLInterfaceType = new GraphQLInterfaceType({
+        name: "IThing",
+        fields() {
+            return {
+                greeting: {
+                    name: "greeting",
+                    type: GraphQLString
+                }
+            };
+        },
+        interfaces() {
+            return [EntityType];
+        }
+    });
+    const DoohickeyType: GraphQLObjectType = new GraphQLObjectType({
+        name: "Doohickey",
+        fields() {
+            return {
+                greeting: {
+                    name: "greeting",
+                    type: GraphQLString,
+                    resolve(source) {
+                        return doohickeyGreetingResolver(source);
```

Comment:
There's no "correct" answer for what this resolver should do. `IThing` and `Entity` each define a `greeting` field and their implementation's differ. `Doohicky` implements both equally so which should it choose?

Note, I think some languages with multiple inheritance leave it to ordering. The ordering of the inheritance becomes meaningful and the first one wins.

We likely need this to become a compile time error and force the user to define an explicit resolver for the type.

Thinking aloud: Another choice would be to start supporting the ability to define interfaces using (abstract?) classes and then only allow _those_ interfaces to have concrete implementations. That would allow us to piggy-back on TypeScript only allowing single inheritance to ensure we always have a single unambiguous implementation for each field. However, it does not play well with functional style, where you want to be able to define your types as object literals or perhaps interfaces themselves.